x-pack/plugins/task_manager/server/task_running/task_runner.ts

```diff
@@ -223,7 +227,9 @@ export class TaskManagerRunner implements TaskRunner {
    * @param id
    */
   public isSameTask(executionId: string) {
-    return executionId.startsWith(this.id);
+    const executionIdParts = executionId.split('::');
+    const executionIdCompare = executionIdParts.length > 0 ? executionIdParts[0] : executionId;
```

Comment:
I think [`String::split()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) will always returns an array with at least one element, in this case, so this could be re-written as:

    const executionIdCompare = executionIdParts[0];

OTOH, you'd have to **know** that, otherwise you might think this is buggy, and the existing code in the PR will actually work fine, so ... not sure if we should really change it :-)